Sources/ATTFindInformationResponse.swift

```diff
@@ -0,0 +1,209 @@
+    public var attributeData: AttributeData
+    
+    public init(attributeData: AttributeData) {
+        
+        self.attributeData = attributeData
+    }
+    
+    public init?(data: Data) {
+        
+        guard data.count >= ATTFindInformationResponse.length else { return nil }
+        
+        let attributeOpcodeByte = data[0]
+        let formatByte = data[1]
+        let remainderData = Data(data.suffix(from: 2))
+        
+        guard attributeOpcodeByte == type(of: self).attributeOpcode.rawValue,
+            let format = Format(rawValue: formatByte),
+            let attributeData = AttributeData(data: remainderData, format: format)
+            else { return nil }
+        
+        self.attributeData = attributeData
+    }
+    
+    public var data: Data {
+        
+        // first 2 bytes are opcode and format
+        return Data([type(of: self).attributeOpcode.rawValue, attributeData.format.rawValue]) + attributeData.data
+    }
+}
+
+// MARK: - Supporting Types
+
+public extension ATTFindInformationResponse {
+
+    public enum Format: UInt8 {
+        
+        /// A list of 1 or more handles with their 16-bit Bluetooth UUIDs.
+        case bit16      = 0x01
+        
+        /// A list of 1 or more handles with their 128-bit UUIDs.
+        case bit128     = 0x02
+        
+        public init?(uuid: BluetoothUUID) {
+            
+            switch uuid {
+            case .bit16: self = .bit16
+            case .bit32: return nil
+            case .bit128: self = .bit128
+            }
+        }
+        
+        public var length: Int {
+            
+            switch self {
+            case .bit16: return 2 + 2
+            case .bit128: return 2 + 16
+            }
+        }
+    }
+}
+
+public extension ATTFindInformationResponse {
+    
+    /// Found Attribute Data.
+    public enum AttributeData {
+        
+        /// Handle and 16-bit Bluetooth UUID
+        case bit16([(UInt16, UInt16)])
+        
+        /// Handle and 128-bit UUIDs
+        case bit128([(UInt16, UInt128)])
+        
+        /// The data's format.
+        public var format: Format {
+            
+            switch self {
+            case .bit16: return .bit16
+            case .bit128: return .bit128
+            }
+        }
+        
+        /// Number of items.
+        public var count: Int {
+            
+            switch self {
+            case let .bit16(value): return value.count
+            case let .bit128(value): return value.count
+            }
+        }
+        
+        /*
+         public static func == (lhs: Data, rhs: Data) -> Bool {
+         
+         switch (lhs, rhs) {
+         
+         case let (.bit16(lhsValue), .bit16(rhsValue)):
+         return lhsValue == rhsValue
+         
+         case let (.bit128(lhsValue), .bit128(rhsValue)):
+         return lhsValue == rhsValue
+         
+         default:
+         return false
+         }
+         }*/
```

Comment:
@colemancda I think we should remove commented code or add a TODO If something is missing